junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/CsvFileArgumentsProvider.java

```diff
@@ -160,4 +161,50 @@ private void advance() {
 
 	}
 
+	private interface Source {
+
+		InputStream open(ExtensionContext context);
+
+	}
+
+	interface InputStreamProvider {
+
+		InputStream openClasspathResource(Class<?> baseClass, String path);
+
+		InputStream openFile(String path);
+
+		default Source classpathResource(String path) {
+			return context -> openClasspathResource(context.getRequiredTestClass(), path);
+		}
+
+		default Source file(String path) {
+			return context -> openFile(path);
+		}
+
+	}
+
+	private static class DefaultInputStreamProvider implements InputStreamProvider {
+
+		private static final DefaultInputStreamProvider INSTANCE = new DefaultInputStreamProvider();
+
+		@Override
+		public InputStream openClasspathResource(Class<?> baseClass, String path) {
+			Preconditions.notBlank(path, "Classpath resource [" + path + "] must not be null or blank");
+			InputStream inputStream = baseClass.getResourceAsStream(path);
+			return Preconditions.notNull(inputStream, () -> "Classpath resource [" + path + "] does not exist");
+		}
+
+		@Override
+		public InputStream openFile(String path) {
+			Preconditions.notBlank(path, "File [" + path + "] must not be null or blank");
+			try {
+				return Files.newInputStream(Paths.get(path));
+			}
+			catch (IOException e) {
+				throw new UncheckedIOException("File [" + path + "] could not be read", e);
```

Comment:
Shouldn't this rather be a `JUnitException` (or subclass thereof)?

Reply:
You're right. I changed it.